src/js.rs

```diff
@@ -27,28 +27,33 @@ thread_local!(
     static RNG_SOURCE: Result<RngSource, Error> = getrandom_init();
 );
 
-pub(crate) fn getrandom_inner(dest: &mut [u8]) -> Result<(), Error> {
+pub(crate) unsafe fn getrandom_inner(mut dst: *mut u8, mut len: usize) -> Result<(), Error> {
     RNG_SOURCE.with(|result| {
         let source = result.as_ref().map_err(|&e| e)?;
 
         match source {
             RngSource::Node(n) => {
-                if n.random_fill_sync(dest).is_err() {
+                let dst = core::slice::from_raw_parts_mut(dst, len);
```

Comment:
Note that strictly speaking this line is unsound, since we can create slice which points to uninitialized memory. Practically speaking, it's fine since we know that no one will read from the created slice, but we still break the Rust rules.

Ideally we would use a pointer-based analog of `random_fill_sync`, but I am not sure whether it's possible to create it. At the very least it could require changes in `js-sys`.

As a temporary workaround we could use a similar approach to the browser branch, i.e. create a temporary buffer initialized with zeros, fill it with random data, and finally copy the random data to the target buffer. Obviously, it will introduce unnecessary overhead which we try to remove with the raw API in the first place.

Reply:
As a better solution, I added initialization of the input buffer with zeros. It's not ideal, but should be tolerable.